packages/back-end/src/services/informationSchema.ts

```diff
@@ -84,25 +85,25 @@ export async function mergeStaleInformationSchemaWithUpdate(
       database.dateCreated =
         staleInformationSchema[correspondingIndex].dateCreated;
     }
-    if (!database.schemas) return;
+    if (!database.schemas || correspondingIndex === -1) return;
     database.schemas.forEach((schema) => {
       const correspondingSchemaIndex = staleInformationSchema[
         correspondingIndex
-      ].schemas.findIndex(
+      ]?.schemas.findIndex(
         (staleSchemaRecord) =>
           staleSchemaRecord.schemaName === schema.schemaName
       );
 
       if (correspondingSchemaIndex > -1) {
         schema.dateCreated =
-          staleInformationSchema[correspondingIndex].schemas[
+          staleInformationSchema[correspondingIndex]?.schemas[
             correspondingSchemaIndex
           ].dateCreated;
       }
       if (!schema.tables) return;
```

Comment:
I think it might make sense to update the early return here as well if `correspondingSchemaIndex === -1`.